src/invoice.ts

```diff
@@ -59,12 +59,18 @@ export default class Invoice {
 
   async verifyPayment(): Promise<boolean> {
     if (!this.verify) throw new Error("LNURL verify not available");
-    const result = await fetch(this.verify);
-    const json = await result.json();
-    if (json.preimage) {
-      this.preimage = json.preimage;
-    }
 
-    return json.settled;
+    try {
+      const result = await fetch(this.verify);
+      const json = await result.json();
+      if (json.preimage) {
+        this.preimage = json.preimage;
+      }
+
+      return json.settled;
+    } catch (error) {
+      console.error("A network error occurred");
```

Comment:
```suggestion
      console.error("failed to check LNURL-verify", error);
```

This doesn't swallow the error (which is not necessarily a network error)